From 4352fc4238b5893796ec6d82193cffc3b845cb6b Mon Sep 17 00:00:00 2001
From: Eric Paris <eparis@redhat.com>
Date: Wed, 10 Sep 2014 14:32:44 -0400
Subject: [PATCH 1/3] KUBE_EXTRA_GOPATH to append to the GOPATH

So you can add your own package to the GOPATH outside of the kube tree
---
 hack/config-go.sh | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/hack/config-go.sh b/hack/config-go.sh
index d49b8a0..70a4355 100644
--- a/hack/config-go.sh
+++ b/hack/config-go.sh
@@ -105,8 +105,13 @@ kube::setup_go_environment() {
     fi
   fi
 
+  GOPATH=${KUBE_TARGET}
+  # include external GOPATH sources
+  if [[ -n ${KUBE_EXTRA_GOPATH:-} ]]; then
+    GOPATH=${GOPATH}:${KUBE_EXTRA_GOPATH}
+  fi
   # Set GOPATH to point to the tree maintained by `godep`.
-  GOPATH="${KUBE_TARGET}:${KUBE_REPO_ROOT}/Godeps/_workspace"
+  GOPATH="${GOPATH}:${KUBE_REPO_ROOT}/Godeps/_workspace"
   export GOPATH
 
   # Unset GOBIN in case it already exists in the current session.
-- 
1.9.3

